pipit/trace.py

```diff
@@ -62,180 +71,227 @@ def _match_events(self):
                 self.events["Event Type"].isin(["Enter", "Leave"])
             ]
 
-            for process in set(enter_leave_df["Process"]):
-                curr_process_df = enter_leave_df.loc[
-                    enter_leave_df["Process"] == process
-                ]
-                for thread in set(curr_process_df["Thread"]):
-                    # filter by both process and thread
-                    filtered_df = curr_process_df.loc[
-                        curr_process_df["Thread"] == thread
+            # list of processes and/or threads to iterate over
+            if "Thread" in self.events.columns:
```

Comment:
this change had to be made in both the _match_event and _match_caller_callee functions, so that it can be used for projections too, which doesn't store a "Thread" column. So we only filter by thread if the column exists.